core/src/main/scala/org/http4s/Entity.scala

```diff
@@ -16,19 +16,64 @@
 
 package org.http4s
 
-import cats.Monoid
+import fs2.{Chunk, Stream}
+import cats._
 import cats.syntax.all._
 
-final case class Entity[+F[_]](body: EntityBody[F], length: Option[Long] = None)
+sealed trait Entity[F[_]] {
```

Comment:
One of the things I've been thinking about lately is whether we could support more efficient file serving. The only way to send a file (or a region of a file) with the current http4s model is to read it from disk, copy it into memory, then write it back out to a socket again. `sendfile` (or `FileChannel.transferTo`) lets you completely circumvent the userspace and copying by writing the file directly to the outgoing socket in kernel space. With the proposed model, I think there is an avenue for more efficiency.

I think we can pass in a `FileChannel` or fs2 `FileHandle` and use that in different ways on different backends. For backends we control (like blaze and ember), we can use `transferTo` to write the file more efficiently. For other backends, we could do the regular pattern of streaming the file in chunks. We know the size of the file so we could do a content-length/chunked/multipart encoding. 

One issue I can see with that is that it can introduce a `Sync`/`Blocker`/`Files` constraint on certain backends (particularly Ember) for file operations. FS2 also doesn't support the `transferTo` operation so we would need to look into feasibility. General thoughts on the approach though?